Cmake Error if BUILD_ARCH is not defined

cmake ..
CMAKE_BUILD_TYPE not set - setting it to RELEASE
Compiling in system Linux-3.12.26-tmb-laptop-1.mga4
CMAKE_BUILD_TYPE: 'RELEASE'
cmake: RELEASE compilation
Building for 
CMake Error at CMakeLists.txt:133 (if):
  if given arguments:

    "STREQUAL" "x86_64"

  Unknown arguments specified


-- Configuring incomplete, errors occurred!

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -130,11 +130,11 @@ include_directories("/usr/include")
 # Library directories
 #
 MESSAGE("Building for ${BUILD_ARCH}")
-if (${BUILD_ARCH} STREQUAL "x86_64")
+if ("${BUILD_ARCH}" STREQUAL "x86_64")
     link_directories("/usr/lib64")
-else (NOT ${BUILD_ARCH} STREQUAL "x86_64")
+else (NOT "${BUILD_ARCH}" STREQUAL "x86_64")
     link_directories("/usr/lib")
-endif (${BUILD_ARCH} STREQUAL "x86_64")
+endif ("${BUILD_ARCH}" STREQUAL "x86_64")
 link_directories("/usr/lib/x86_64-linux-gnu")
 #
 # Enabling test harness

